### PR TITLE
Split unsquashed merge tests into dedicated test methods

### PR DESCRIPTION
**Pull request by dimas-b**
This is in preparatiuon for API v2, which will not allow unsquashed merges.

### CODE REVIEW

File: servers/jax-rs-tests/src/test/java/org/projectnessie/jaxrs/AbstractRestSecurityContext.java [hunk 1]
```diff
@@ -103,32 +98,46 @@ public void committerAndAuthor(
...
 
     getApi()
         .transplantCommitsIntoBranch()
-        .fromRefName(withSecurityContext.getName())
-        .hashesToTransplant(
-            Arrays.asList(noSecurityContext.getHash(), withSecurityContext.getHash()))
+        .fromRefName(main.getName())
+        .hashesToTransplant(hashesToTransplant)
         .branch(transplant)
         .keepIndividualCommits(true)
```
Comment (ajantha-bhat) [inline servers/jax-rs-tests/src/test/java/org/projectnessie/jaxrs/AbstractRestSecurityContext.java:142]: So, this test case name should be `committerAndAuthorTransplantUnsquashed`?

Also, are we missing the squashed `committerAndAuthorTransplant` test case?

**Review by snazy — APPROVED** [review]
The refactoring itself is fine.
Side note: this test class is called `AbstractRestSecurityContext` - not `AbstractRestMergeTransplant`

File: servers/jax-rs-tests/src/test/java/org/projectnessie/jaxrs/AbstractRestSecurityContext.java (hunk 1 above)
Comment (dimas-b) [reply servers/jax-rs-tests/src/test/java/org/projectnessie/jaxrs/AbstractRestSecurityContext.java:142]: Transplant is always unsquashed, AFAIK

**Comment by dimas-b** [comment]
> Side note: this test class is called AbstractRestSecurityContext - not AbstractRestMergeTransplant

Yeah, I did not pay attention to that :facepalm: maybe later? :)

**Pull request merged**